### ISSUE TITLE
[Discussion] Tech Sharing 11 Jun

### ISSUE THREAD

**Issue by rjchow** [OpenCerts/opencerts-website#327]
# Agenda
Generally get the team in sync on how to collaborate.
Feel free to update this issue with items you would like to discuss, or points to clarify on before meeting.

### Workflow
- Pull Request guidelines
  - Using hub to convert issues to PRs
  - Writing a good PR description
  - What code changes should be in a PR (size of PRs)
  - Resolving issues in a PR (i.e only the reviewer who opens an issue should close it)
  - Closing a PR
- Regression tests on opencerts-website

### Boilerplate Stuff
- Changelog management (semantic release, commit lint, angular commit message style)
- Typescript
- Documentation.js

**Comment by rjchow** [comment]
## Writing a good PR title and description
A good PR description should:
### Explain the rationale of the issue being resolved 
This PR could be read by someone 6 months or a year from now who has completely no context of the issue - he may have gotten here by looking up the Git commit that resulted in this line of code, and is wondering why it was written this way.

More immediately, this PR might be being reviewed by one of us who was not present when the decision was made to develop this feature. Having a good PR description sets the context for the reviewer and makes it much easier for anyone to pick it up for review.

The PR title should be the same as the commit message summary, and the PR Description should be the same as the commit message body.
(To make this easier, a PR should be converted from the issue that was lodged, and already has the details. More on how to do this below)

### Explain the approach taken to resolve the issue
Briefly explain what changes were necessary and the impact of the changes (if any). Also document any difficulties faced that resulted in design decisions being made. This way it cuts down on the back-and-forth of being questioned why certain things were done, and also relieves the reviewer from having to feel like he's an asshole for nitpicking on everything ;-)


### Detail follow up items
If there are any outstanding issues related to this that was not solved in this PR, create those issues before closing this one, and link back to this. If the issue is too large to solve in one PR, make multiple PRs and link back to the original issue. If this PR closes another issue, link back to that issue so that they know it's closed. You can use Github's keywords to help with this: [Closing issues using keywords - GitHub Help](https://help.github.com/en/articles/closing-issues-using-keywords)


### PR Titles
PR titles should follow the Angular commit message convention as we will be introducing changelog automation at some point in the future. [Angular Commit Format Reference Sheet · GitHub](https://gist.github.com/brianclements/841ea7bffdb01346392c). The PR title should also be the same as the commit message summary.

### Some examples:

###### #325 
**[Feat] Creating a 'partners' page for linking to our partners**
This issue was created to advertise our partners' services so that someone interested in engaging their services will not have to contact us.

Since the registry page serves a similar purpose, the partners page will copy from the registry page component. At a later date if we need to copy it yet again we should probably consider generalising the component. (rule of three)


###### #314 
**[Feat] Adding Google Analytics trigger to demo drop function**
This issue was created because we would like to track the analytics data for the demo feature as it will help us quantify the impact of the demo feature.

There already exists the analyticsEvent function which makes this story a simple addition of the trigger as an onDragEnd event handler on the demo certificate component.

To check that this feature works, you can turn on debug logging and confirm that the analyticsEvent is indeed fired.

###### #270 
**[Fix] Fix browser back button causing unregistered instution warning**
This issue was created because the application state becomes inconsistent after the back button is used.

Steps to reproduce:
  1. Load a valid cert for viewing
  2. Use the browser back button to go back to main page
  3. Load the same cert for viewing
  4. Unregistered institution warning is displayed

[Video of bug](https://github.com/OpenCerts/opencerts-website/files/3175847/Peek.2019-05-14.12-34.zip)

This bug was determined to be caused by [setNewWeb3 failing](https://github.com/OpenCerts/opencerts-website/blob/891f0c016f5c6f441c41664fcf441aa044a91304/src/services/web3/getWeb3.js#L92). The problem is resolved by removing  web3Instance.currentProvider.stop() as it was causing the function to error. This line is safe to remove as it was only used previously when we used the ledger-usb provider.

A regression test that replicates this error was introduced as well.

**Comment by rjchow** [comment]
## PR Etiquette

- Be nice and polite to each other in reviews
- Don't take it personally when feedback is given or questions are asked, nobody is out to shame or annoy anyone else here
- The person who worked on the PR should merge the PR (if multiple just pick someone or the most recent)
- Merge a PR only after the reviewer has labelled it ready for merge
- Don't merge a PR if there are unresolved feedback issues
- The reviewer or whomever that opened the feedback issue should resolve it, not the developer working on the PR

**Comment by rjchow** [comment]
## Using Hub to convert issues to pull requests

There's no way to do this from the GitHub web interface but you can use the Hub cli utility to convert an existing issue to a pull request. [GitHub - github/hub: A command-line tool that makes git easier to use with GitHub.](https://github.com/github/hub)

The command is run when you're on the branch you would like to make the PR from, and suffixed with the parameter `-i <issue number>`

E.g:
`hub pull-request -i 240`
